### ISSUE TITLE
Call of Duty: World at War (10090)

### ISSUE THREAD

**Issue by Aerocatia** [ValveSoftware/Proton#360]
Has been working really well, there are no issues other than the expected performance hit of dx9 translation.
I did not even bother with punkbuster though as you need a 3rd party fix for that even on windows, the game still works perfectly without it sans joining PB servers.

**Comment by xDShot** [comment]
Keep in mind that first time you launch it after downloading it, don't cancel DirectX installation. Otherwise, you get missing sounds.

**Comment by Aerocatia** [comment]
Also that caveat should be fixed by https://github.com/ValveSoftware/wine/issues/2

**Comment by kisak-valve** [comment]
Crackling sound in all games

Issue transferred from https://github.com/ValveSoftware/Proton/issues/1395.
@PureTryOut posted on 2018-09-11T15:22:19:

# Compatibility Report
- Name of the game with compatibility issues: all games, but some worse than others, using COD:WAW for this 
- Steam AppID of the game: 10090

## System Information
- GPU: RX480
- Driver/LLVM version: Mesa 18.2/6.0.1
- Kernel version: 4.18.7
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/PureTryOut/ffad4816866322e165dd4047f009c4f9
- Proton version: any, but right now 3.7-6 Beta

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms 

Sound crackling in every game. It doesn't always happen, and for some games it happens more than others (for example in Call of Duty: World at War it happens about 90% of the time, where with GTA5 it happens about 20% of the time and only after playing for a bit).

I've uploaded a sound fragment from a game of COD:WAW where you can hear it clearly (a .wav in a .zip).
[cod-crackling-sound.zip](https://github.com/ValveSoftware/Proton/files/2371556/cod-crackling-sound.zip)

## Reproduction

1. Launch COD:WAW
2. Open the server browser, you'll hear explosions in the background which should crackle
3. If the sound doesn't crackle, you've caught the lucky 10%, and you have to restart the game

**Comment by mirh** [comment]
I wonder how many games couldn't be basically summed up with "wait for FAudio"

**Comment by Aerocatia** [comment]
Are you a GPU passthrough user by any chance @PureTryOut?
I have the exact same audio issues in some games on wine/proton including this one but it only happens if a guest VM has been started that boot, apparently it is a known iommu bug but no one knows the exact cause.
If no vfio VM is ever started then audio is fine in wine and it's forks.

**Comment by PureTryOut** [comment]
I do not, I see no reason to with Proton. Anyways, it's already resolved (a workaround) in #120.

**Comment by Holzhaus** [comment]
I have no issues, everyhing works fine (even multiplayer).

**Comment by Seltyk** [comment]
The performance hit on W@W is nasty (my Vega 56 can only get 45FPS in zombies solo, min settings).

UPDATE: trying to run the game using `PROTON_USE_D9VK=1 %command%`, with or without T4M, runs at the same low framerate as before, except this time it crashes upon loading a map. Also, removed T4M comment for being irrelevant.

**Comment by kisak-valve** [comment]
Call of Duty World at War crash with Proton 4.11.11 ( does not use DXVK because DirectX installation) 

Issue transferred from https://github.com/ValveSoftware/Proton/issues/3351.
@iWeaker4you posted on 2019-12-22T21:29:31:

# Compatibility Report
- Name of the game with compatibility issues: Call of Duty World at War
- Steam AppID of the game: 10090

## System Information
- GPU: GTX1050
- Driver/LLVM version: NVIDIA 440.43.01
- Kernel version: 5.4.6-arch1-1
- Proton version: 4.11.11

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-10090.log](https://github.com/ValveSoftware/Proton/files/3993024/steam-10090.log)

## Symptoms 

DirectX installation causes the game not to start with the D9VK dll (DXVK 1.5 ), i use the `PROTON_USE_D9VK=1 %command%`, it works but in certain points crash , the first time crashing at 5 minutes, in the second attempt crashing at 20min.

**Comment by kisak-valve** [comment]
Hello @iWeaker4tyou, the log has a bunch of `err:   DxvkMemoryAllocator: Memory allocation failed`, which hints that you encountered https://github.com/doitsujin/dxvk/issues/1100#issuecomment-552784194. You might get lucky trying `PROTON_FORCE_LARGE_ADDRESS_AWARE=1 PROTON_USE_D9VK=1 %command%` in the game's launch options.

**Comment by iWeaker** [comment]
@kisak-valve  thanks, its works very well with PROTON_FORCE_LARGE_ADDRESS_AWARE=1, no crash anymore.

**Comment by kisak-valve** [comment]
Call of Duty World at War Crashes with DXVK

Issue transferred from https://github.com/ValveSoftware/Proton/issues/4256.
@billybob32 posted on 2020-10-12T01:12:09:

# Compatibility Report
- Name of the game with compatibility issues: Call of Duty: World at War
- Steam AppID of the game: 10090

## System Information
- GPU: AMD R5 M430
- Driver/LLVM version: Mesa 20.0.5
- Kernel version: 5.4.0-48-generic
- Link to full system information report as https://gist.github.com/billybob32/a5da6bab842d3bdc4ebb1a9891a0a179
- Proton version: 5.0-9

## I confirm:
- [Yes] that I haven't found an existing compatibility report for this game.
- [Yes] that I have checked whether there are updates for my system available.

`<Proton log omitted, please see #4256>`

## Symptoms
Game just doesn't run. It runs for a bit, then crashes. used to crash my whole pc. Now it just crashes the game.

## Reproduction
I just have to launch it on steam, it crashes. Forcing it to not use DXVK with PROTON_USE_WINED3D=1 %command% fixes the issue.

**Comment by kisak-valve** [comment]
Hello @billybob32, since this issue report was last commented on, PROTON_FORCE_LARGE_ADDRESS_AWARE was enabled by default with Proton 5.0, so there's no point in testing that. It would have been nice to have had the extended diagnostic information in your system info to confirm that you have a working 32 bit vulkan driver stack. Please verify that `mesa-vulkan-drivers` and `mesa-vulkan-drivers:i386` are installed with something like `apt policy mesa-vulkan-drivers mesa-vulkan-drivers:i386`.

Also, it may be worthwhile testing a newer mesa build from a PPA like [oibaf](https://launchpad.net/~oibaf/+archive/ubuntu/graphics-drivers) or [kisak-mesa](https://launchpad.net/~kisak/+archive/ubuntu/kisak-mesa).

**Comment by Seltyk** [comment]
I've noticed that while the game runs well enough, some audio is missing. Repairing barriers and opening doors in zombies are particularly common offenders. I've confirmed this on all versions since 4.11, including 5.13-5 with the latest beta SLC Soldier

**Comment by trymeouteh** [comment]
I find the game runs perfectly with and without mods using Proton 4.11-13.

However when using the latest version of Proton 8.0-4, the game runs fine without mods, but if not all zombie maps mods will not load and cause the game to either crash or freeze when loading the zombie map.

Here are the logs for this using Proton 8.0-4 loading [Nacht Der Untoten Reimagined (Version 2.1)](https://www.ugx-mods.com/forum/map-releases/29/nacht-der-untoten-reimagined-version-2-1/7467) mod.

[steam-10090.log](https://github.com/ValveSoftware/Proton/files/13245664/steam-10090.log)

**Comment by mirh** [comment]
What about 5.0, 5.13, 6.3 or 7.0?

**Comment by trymeouteh** [comment]
I get the same results with 7.0 as I do with 8.0-4.

**Comment by bigjoe54321** [comment]
## System Information
- GPU:NVIDIA GeForce RTX 3070
- Video driver version: 610.43.02
- Kernel version: 7.1.1-2-cachyos
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/bigjoe54321/23cca11b98f607d4406ad2104be30f96
- Proton version: Experimental

## I confirm:
- [ ] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms 
Call of Duty: World at War immediately stops after pressing the play button in Steam when the game is installed to my HDD I mounted to `/mnt/7200RPM_HDD`.

## Reproduction

1. On Linux, install Call of Duty: World at War to a mounted storage drive that does not contain your `/home` mount point
2. Force Call of Duty: World at War to use the Proton Experimental compatibility tool
3. Play Call of Duty: World at War
4. Watch the game immediately stop and the green play button reappear
[steam-10090.log](https://github.com/user-attachments/files/29438067/steam-10090.log)

**Comment by kisak-valve** [comment]
Hello @bigjoe54321, the noexec lines in your Proton log point towards there being an issue with the execute permission on the partition used with your secondary Steam library folder.

**Comment by bigjoe54321** [comment]
Thanks for your reply, @kisak-valve. I tried the following fixes, but the main problem still remains.

- Run `fd -H -tf -e exe . . -X chmod +x -c` in `/mnt/7200RPM_HDD/SteamLibrary/steamapps/compatdata`
- Run `fd -H -tf -e dll . . -X chmod +x -c` in `/mnt/7200RPM_HDD/SteamLibrary/steamapps/compatdata`
- Run `fd -H -tf -e exe . . -X chmod +x -c` in `/mnt/7200RPM_HDD/SteamLibrary/steamapps/common/Call of Duty World at War`
- Run `fd -H -tf -e dll . . -X chmod +x -c` in `/mnt/7200RPM_HDD/SteamLibrary/steamapps/common/Call of Duty World at War`

Here's a new log:
[steam-10090.log](https://github.com/user-attachments/files/29440461/steam-10090.log)

**Comment by bigjoe54321** [comment]
I think I figured out what you meant, because of what this command showed me. I'll learn how to remove `noexec`. I just learned that `noexec` is not in my fstab file.

```fish
❯ mount | rg "btrfs.*noexec"
/dev/sdb1 on /mnt/5400RPM_HDD type btrfs (rw,nosuid,nodev,noexec,noatime,compress=zstd:3,space_cache=v2,commit=120,subvolid=5,subvol=/,user)
/dev/sdc1 on /mnt/7200RPM_HDD type btrfs (rw,nosuid,nodev,noexec,noatime,compress=zstd:3,space_cache=v2,commit=120,subvolid=5,subvol=/,user)
```
### Edit:
I found this.
https://man7.org/linux/man-pages/man8/mount.8.html#:~:text=This%20option%20implies%0A%20%20%20%20%20%20%20%20%20%20%20the%20options%20noexec

>       users
>           Allow any user to mount and to unmount the filesystem, even
>           when some other ordinary user mounted it. This option implies
>           the options noexec, nosuid, and nodev (unless overridden by
>           subsequent options, as in the option line
>           users,exec,dev,suid)

It seems the problem will be fixed after I make the following change to my fstab file:
```diff
-UUID=4de52f34-a880-40e6-8131-c316a9c2a554   /mnt/7200RPM_HDD   btrfs   noatime,compress=zstd,commit=120,user             0 0
+UUID=4de52f34-a880-40e6-8131-c316a9c2a554   /mnt/7200RPM_HDD   btrfs   noatime,compress=zstd,commit=120,user,exec             0 0
```

### Edit 2:
Yes, adding the `exec` option to that line in fstab fixed the problem 🙂 Thanks again for your help, @kisak-valve.
